### ISSUE TITLE
OAuth authorization issue

### ISSUE THREAD

**Issue by ianovi** [baileyherbert/envato.js#11]
Hello Bailey!

I've got the error when using OAuth:

```
Authentication failed: OAuthError: Received an HTTP error
    at OAuth.<anonymous> (.../node_modules/envato/dist/clients/oauth.js:51:23)
    at Generator.next (<anonymous>)
    at fulfilled (.../node_modules/envato/dist/clients/oauth.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  http: BadRequestError: Bad Request
      at HttpClient._getHttpError (.../node_modules/envato/dist/helpers/http.js:111:30)
      at HttpClient.<anonymous> (.../node_modules/envato/dist/helpers/http.js:43:29)
      at Generator.next (<anonymous>)
      at fulfilled (.../node_modules/envato/dist/helpers/http.js:5:58)
      at processTicksAndRejections (node:internal/process/task_queues:96:5) {
    code: 400,
    response: {
      error: 'invalid_request',
      error_description: 'No authorization header passed'
    }
  }
}
```

Issue coming from line:

`const client = await oauth.getClient(code);`

from your example:

```
// This is an example express route to handle the request
app.get('/authenticate', async function(req, res) {
    try {
        // Fetch the code from the URL
        const code = req.query.code;

        // Authenticate the user via the API
    	const client = await oauth.getClient(code);
        const username = await client.private.getUsername();

        // You can also get their User ID for data storage
        const { userId } = await client.getIdentity();

        res.send(`Hello, ${username}!`);
    }
    catch (error) {
        console.error('Authentication failed:', error);
        res.status(500).send('Something broke! Try again.');
    }
});
```

My suggestion, how to fix it in attach:
[Fix_for_OAuth_authorization_flow.patch.zip](https://github.com/baileyherbert/envato.js/files/9092029/Fix_for_OAuth_authorization_flow.patch.zip)

Thanks in advance.

**Comment by baileyherbert** [comment]
Hi @HotTemplate, thanks for reporting this and apologies for my slow response time! 😅
I've released a fix for this in `v3.0.7`, let me know if you're still having issues after updating. Cheers!

**Comment by ianovi** [comment]
Yes, it's works now. Thanks a lot.
